diff --git a/xbin/zorba3-0.xqy b/xbin/zorba3-0.xqy
index ba07c2e..481950f 100644
--- a/xbin/zorba3-0.xqy
+++ b/xbin/zorba3-0.xqy
@@ -212,7 +212,11 @@ let $marcxml :=
         let $json := http:get($marcxmluri)
         return parsexml:parse($json("body")("content"), <parseoptions:options/>)
     else
-        let $raw-data as xs:string := file:read-text($marcxmluri)
+        let $raw-data :=
+            if ( fn:starts-with($marcxmluri, "raw:" ) ) then
+                fn:substring($marcxmluri, 5)
+            else
+                file:read-text($marcxmluri)
         let $mxml := parsexml:parse(
                     $raw-data, 
                     <parseoptions:options />
